File: FootballExplorer_java/app/src/main/java/fr/a42consulting/footballexplorer/View/PlayersListAdapter.java
package fr.a42consulting.footballexplorer.View;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import fr.a42consulting.footballexplorer.Model.PlayersListPlayer;
import fr.a42consulting.footballexplorer.R;

public class PlayersListAdapter extends ArrayAdapter<PlayersListPlayer> {
    private Context context;

    public PlayersListAdapter(Context context, int resource) {
        super(context, resource);
        this.context = context;
    }

    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View v = inflater.inflate(R.layout.list_player_element, null);

        PlayersListPlayer player = getItem(position);
        ImageView playerImage = v.findViewById(R.id.playerImage);
        TextView playerName = v.findViewById(R.id.playerName);
        TextView playerRole = v.findViewById(R.id.playerRole);
        TextView playerBirth = v.findViewById(R.id.playerBirth);
        TextView playerPrice = v.findViewById(R.id.playerPrice);

        Glide.with(context).load(player.getImageUrl()).into(playerImage);
        playerName.setText(player.getName());
        playerBirth.setText(player.getBirth());
        playerRole.setText(player.getRole());
        playerPrice.setText(player.getPrice());

        return v;
    }
}

File: FootballExplorer_kotlin/app/src/main/java/fr/a42consulting/footballexplorer/view/adapter/PlayersListAdapter.kt
package fr.a42consulting.footballexplorer.view.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.bumptech.glide.Glide

import fr.a42consulting.footballexplorer.model.PlayersListPlayer
import fr.a42consulting.footballexplorer.R
import kotlinx.android.synthetic.main.list_player_element.view.*

class PlayersListAdapter(private val mContext: Context, val items: ArrayList<PlayersListPlayer>) : RecyclerView.Adapter<PlayersListAdapter.ViewHolder>() {
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val player = items[position]
        Glide.with(mContext).load(player.imageUrl).into(holder.playerImage)
        holder.playerName.text = player.name
        holder.playerRole.text = player.role
        holder.playerBirth.text = player.birth
        holder.playerPrice.text = player.price
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return PlayersListAdapter.ViewHolder(LayoutInflater.from(mContext).inflate(R.layout.list_player_element, parent, false))
    }

    class ViewHolder (itemView: View): RecyclerView.ViewHolder(itemView) {
        val playerImage = itemView.playerImage!!
        val playerName = itemView.playerName!!
        val playerRole = itemView.playerRole!!
        val playerBirth = itemView.playerBirth!!
        val playerPrice = itemView.playerPrice!!
    }
}
